lib/couchbaseX.js

```diff
@@ -72,18 +71,26 @@ class Couchbase extends NoSQL {
       settings.bucket.password = '';
     }
     // Cluster.
-    this._cluster = Promise.resolve(new couchbase.Cluster(settings.cluster.url, settings.cluster.options))
-      .then(utils.promisifyAllResolveWithReturn);
+    if (parseInt(settings.version, 10) < 5) {
+      // Coubase version < 5
+      this._cluster = Promise.resolve(new couchbase.Cluster(settings.cluster.url, settings.cluster.options))
+        .then(utils.promisifyAllResolveWithReturn);
+    } else {
+      // Coubase version >= 5, as default
+      const cluster = new couchbase.Cluster(settings.cluster.url, settings.cluster.options);
+      cluster.authenticate(settings.cluster.username, settings.cluster.password);
```

Comment:
This line is the only difference between the `if` and `else`, could you try to remove the duplicated codes?